WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -947,11 +947,15 @@ private static int GetInputTarget(WasabiRandom rnd)
 		return targetInputCount;
 	}
 
-	private async Task<IEnumerable<TxOut>> ProceedWithOutputRegistrationPhaseAsync(uint256 roundId, ImmutableArray<AliceClient> registeredAliceClients, CancellationToken cancellationToken)
+	private async Task<(bool, IEnumerable<TxOut>)> ProceedWithOutputRegistrationPhaseAsync(uint256 roundId, ImmutableArray<AliceClient> registeredAliceClients, CancellationToken cancellationToken)
 	{
 		// Waiting for OutputRegistration phase, all the Alices confirmed their connections, so the list of the inputs will be complete.
 		var roundState = await RoundStatusUpdater.CreateRoundAwaiterAsync(roundId, Phase.OutputRegistration, cancellationToken).ConfigureAwait(false);
 		var roundParameters = roundState.CoinjoinState.Parameters;
+
+		// Detects if our own inputs are registered by someone else.
+		var shouldRegisterOutputs = ShouldRegisterOutputs(registeredAliceClients, roundState);
```

Comment:
I would name it `var maliciousInputsDetected`. It makes more sense to name something according to the logic of how it was generated.